prov/rxd/src/rxd.h

```diff
@@ -215,13 +215,16 @@ struct rxd_ep {
 	struct dlist_entry active_peers;
 	struct dlist_entry rts_sent_list;
 	struct dlist_entry ctrl_pkts;
-
-	struct rxd_peer **peers;
+	
+	struct index_map peers_idm;
 };
 
 static inline struct rxd_peer *rxd_peer(struct rxd_ep *ep, fi_addr_t rxd_addr)
 {
-	return ep->peers[rxd_addr];
+	fastlock_acquire(&ep->util_ep.lock);
+	return ofi_idm_lookup(&ep->peers_idm, rxd_addr);
+	fastlock_release(&ep->util_ep.lock);
+
```

Comment:
I don't think you need a lock around the lookup call.  It's safe to have though.

Reply:
The rxd_peer interface is sometimes/not always called by functions which hold the util_ep.lock( this was causing the CI failures). Would it be better to call something like a fastlock_tryacquire instead or would it be better to remove the lock around the lookup call.

Reply:
It's probably safe to keep this as-is.  We can potentially remove this lock in the future as an optimization.  We need a lock around insert/remove, but lookup should be safe without a lock